From 03d0a9fc2a2fdead6648877bb193bbf49047113f Mon Sep 17 00:00:00 2001
From: "Nguyen Van Linh [FGA.BU13]" <LinhNV25@fsoft.com.vn>
Date: Tue, 10 Apr 2018 08:28:07 +0700
Subject: [PATCH 271/628] spi: rspi: Fix SSLKP last SPCMD0

    commit <a475ddac3fd619ff0741186acfb819bba0fc4376>
    from git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-backport.git

    If SSLKP(SSL Signal Level Keeping) is 1, chip select will not be
    negated until SPCR.CPE is set to 0. SSLKP is set to 0 to last SPCMD.

Signed-off-by: Hisashi Nakamura <hisashi.nakamura.ak@renesas.com>
Signed-off-by: khanhluu <khanh.luu.xw@rvc.renesas.com>
Signed-off-by: Nguyen Van Linh [FGA.BU13] <LinhNV25@fsoft.com.vn>
Signed-off-by: vietn <vietn@fsoft.com.vn>
---
 drivers/spi/spi-rspi.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/spi/spi-rspi.c b/drivers/spi/spi-rspi.c
index d489358..90b5e1c 100644
--- a/drivers/spi/spi-rspi.c
+++ b/drivers/spi/spi-rspi.c
@@ -988,6 +988,10 @@ static int qspi_setup_sequencer(struct rspi_data *rspi,
 	}
 	if (i) {
 		/* Set final transfer data length and sequence length */
+		rspi_write16(rspi,
+			rspi_read16(rspi, RSPI_SPCMD(i - 1)) &
+			~SPCMD_SSLKP,
+			RSPI_SPCMD(i - 1));
 		rspi_write32(rspi, len, QSPI_SPBMUL(i - 1));
 		rspi_write8(rspi, i - 1, RSPI_SPSCR);
 	}
-- 
2.7.4

